cmd/pulumi-test-language/providers/secret_provider.go

```diff
@@ -0,0 +1,317 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package providers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/blang/semver"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+)
+
+// A small provider with a single resource "Resource" that has a single field "Data" which is a type with some
+// primitive fields and a field to another nested type "InnerData" which is a type with some primitive fields,
+// these fields are marked secret at various levels.
+// The provider also has an invoke that also takes secret inputs.
```

Comment:
I think this comment was copied from another provider

Reply:
Ah yeh it was, I did tweak it when first writing (see the marked secret bit) but forgot to update it when I changed the schema a bit more.